Kronos-Shifts-Connector/Deployment/azuredeploy.json

```diff
@@ -587,18 +585,53 @@
             ]
         },
         {
-            "name": "[variables('keyVaultName')]",
             "type": "Microsoft.KeyVault/vaults",
-            "apiVersion": "2016-10-01",
+            "apiVersion": "2018-02-14",
+            "name": "[variables('keyVaultName')]",
             "location": "[parameters('location')]",
+            "dependsOn": [
+                "[concat('Microsoft.Web/sites/', variables('apiAppName'))]",
+                "[concat('Microsoft.Web/sites/', variables('configAppName'))]"
+            ],
+            "tags": {
+                "displayName": "KeyVault"
+            },
             "properties": {
-                "sku": {
-                    "family": "A",
-                    "name": "Standard"
-                },
+                "enabledForDeployment": "true",
```

Comment:
Can you explain why these need to be enabled? enabledForDeployment, enabledForTemplateDeployment, enabledForDiskEncryption.

Reply:
@aosolis - Having enabledForDeployment = "true" means that the deployment of the Azure KeyVault can be accessed by the Azure AD Application registration. Then having the flag of enabledForTemplateDeployment = "true" allows for the Azure KeyVault to be created through ARM Templates. See below: 

![image](https://user-images.githubusercontent.com/5623947/76794348-0bd3df00-6784-11ea-88ae-4f86aac276a2.png)

Reply:
That doesn't sound right. Read that page again. https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-tutorial-use-key-vault
enabledForTemplateDeployment = another template deployment can access secrets stored in the key vault

And enabledForDeployment seems to be about granting VMs access to key vault.

Reply:
@coderkrishna Please justify these

Reply:
@aosolis - The enabledForTemplateDeployment = true is allowing for ARM Template deployments. This needs to be enabled because I want the ARM Template to be able to add entries into the KeyVault, and also the KeyVault is being used by both the API and Config app projects.

The enabledForDeployment should be set to false because from the previous discussion, if a Virtual Machine is being created, then the VM can be able to retrieve secrets from the Azure KeyVault. In this case, as there are no VMs, this flag can be set to false.

The enabledForVolumeEncryption will be set to false because there are no virtual disks being used. I will be having these changes reflected properly

Reply:
@coderkrishna 
https://github.com/Azure/azure-quickstart-templates/blob/master/201-key-vault-secret-create/azuredeploy.json
implies that you can set a secret even if enabledForTemplateDeployment  is false.
Note that the documentation says "specify whether Azure Resource Manager is **permitted** to retrieve secrets from the key vault."

Reply:
> and also the KeyVault is being used by both the API and Config app projects.

What does this have to do with enabledForTemplateDeployment?